tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/LanguageRepoService.cs

```diff
@@ -153,23 +176,117 @@ protected async Task<CLICheckResponse> ValidateChangelogCommonAsync(string packa
+                "-File", scriptPath, 
+                "-SettingsPath", settingsPath,
+                "-ScanPaths", packagePath,
+            };
+
+            // Use a longer timeout for README validation - 10 minutes should be sufficient as it may need to install doc-warden
+            var timeout = TimeSpan.FromMilliseconds(600_000); // 10 minutes
+            var processResult = await _processHelper.Run(new(command, args, timeout: timeout, workingDirectory: packagePath), ct: default);
+
+            return CreateResponseFromProcessResult(processResult);
+        }
+        catch (Exception ex)
+        {
+            return new CLICheckResponse(1, "", $"Unhandled exception: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Common spelling check implementation that works for most Azure SDK languages.
+    /// Uses cspell directly to check spelling in the package directory.
+    /// </summary>
+    /// <param name="packagePath">Absolute path to the package directory</param>
+    /// <returns>CLI check response containing success/failure status and response message</returns>
+    protected async Task<CLICheckResponse> CheckSpellingCommonAsync(string packagePath)
+    {
+        try
+        {
+            if (!Directory.Exists(packagePath))
+            {
+                return new CLICheckResponse(1, "", $"Package path does not exist: {packagePath}");
+            }
+
+            // Find the SDK repository root by looking for common repository indicators
+            var packageRepoRoot = _gitHelper.DiscoverRepoRoot(packagePath);
+            if (string.IsNullOrEmpty(packageRepoRoot))
+            {
+                return new CLICheckResponse(1, "", $"Could not find repository root from package path: {packagePath}");
+            }
+
+            // Construct the path to the cspell config file
+            var cspellConfigPath = Path.Combine(packageRepoRoot, ".vscode", "cspell.json");
+            
+            if (!File.Exists(cspellConfigPath))
+            {
+                return new CLICheckResponse(1, "", $"Cspell config file not found at expected location: {cspellConfigPath}");
+            }
+
+            // Convert absolute path to relative path from repo root
+            var relativePath = Path.GetRelativePath(packageRepoRoot, packagePath);
+
+            // Build arguments for npx cspell lint
+            var args = new[] {
+                "cspell", "lint",
+                "--config", cspellConfigPath,
+                "--root", packageRepoRoot,
+                $"./{relativePath}/**"
+            };
+
+            var processResult = await _processHelper.Run(new("npx", args, workingDirectory: packageRepoRoot), ct: default);
+            return CreateResponseFromProcessResult(processResult);
         }
         catch (Exception ex)
         {
-            stopwatch.Stop();
             return new CLICheckResponse(1, "", $"Unhandled exception: {ex.Message}");
         }
         finally
```

Comment:
I can't seem to leave a comment on the correct line, but what does this line mean?

```csharp
finally {
  await Task.CompletedTask; // Make this async for consistency
}
```

Should this finally even exist? Isn't this just the default behavior?

Reply:
it should be removed this was pre- processHelper being an async class